### ISSUE TITLE
Interpreting results.png after training

### ISSUE THREAD

**Issue by wangp22** [ultralytics/ultralytics#21127]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi,

Can you please explain how to interpret the various losses in results.png? I know they are all plotted against epoch number. But how does one know if the curves are good? I think smooth curves are idea whereas spikes means instability or overtraining.

<img width="712" alt="Image" src="https://github.com/user-attachments/assets/f6648368-2d14-436f-84d4-1c91ddc6667a" />

I also need help understanding box loss, cls loss, and dfl loss. I understand precision, recall, and mAP50 and mAP95, although I'm not sure what the (B) means.

BTW, are these metrics averaged over all classes?

Thanks

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wangp22, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response to help you get started while an Ultralytics engineer reviews your question.

For in-depth explanations and visual guides, you may find the [Docs](https://docs.ultralytics.com/) especially helpful, including many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. You might find answers to many common questions there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) if you haven't already, as this helps us debug quickly.

For custom training ❓ questions like this, please share as much info as possible: example images, training logs, and confirm you’ve reviewed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our community where you prefer! For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. For more in-depth discussions, check out [Discourse](https://community.ultralytics.com/). Or join the threads on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Make sure you're using the latest `ultralytics` package and [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). This ensures you're seeing the latest results and features:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in these up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

An Ultralytics engineer will review your question and provide further guidance soon. Thank you for your patience and for being part of the Ultralytics community!

**Comment by zshar7** [comment]
Hello there!
There is no way to say if your graphs are looking "good" as this is subjective from task to task.
A quick look at this seems that you might try and keep training past 100th epoch because both your train and validation loss curves look to be decreasing and might have the potential to decrease more (which is good). What you want to look for is when the loss curves start to plateau or when they increase (particularly when the validation loss increases whilst training loss decreases: that is overfitting! You definitely do not want that).
(B) mentions the boxes and the metrics pertaining to your bounding box. I assume this is an uncut image and that your task was detection with a simple bounding box. It might be safe to ignore unless you are working with segmentation.
I am pretty sure the blue curve is averages across classes.
I will take the time to explain the loss metrics but I highly encourage looking at the Ultralytics Documentation- [this is a really good resource.](https://docs.ultralytics.com/guides/model-evaluation-insights/#how-do-i-access-yolo11-model-evaluation-metrics-in-python) I am fairly new to coding and a high schooler and their tutorials and guides are very easy to understand!
Throughout training, YOLO outputted loss metrics of the best model in every epoch. YOLO measures four loss metrics: bounding box loss (Box Loss), classification loss (CLS loss), and Distribution Focal Loss (DFL). Box Loss measures image area misidentified. CLS Loss determines overall incorrect predictions made by model. DFL works to help class imbalance and directly optimizes a distribution of bounding box borders. 
Again highly suggest looking into the resource I provided. Your training looks phenomenal keep up the good work!

**Comment by glenn-jocher** [comment]
@zshar7 provided an excellent overview! To add a bit more context: the loss curves you're seeing represent the training objective that the model optimizes during learning, while the metrics (precision, recall, mAP) evaluate real-world performance.

For your specific losses: box loss measures localization accuracy, cls loss handles classification errors, and dfl loss (Distribution Focal Loss) improves bounding box regression by treating it as a classification problem over distance bins. The (B) indeed indicates these are box-related metrics for detection tasks.

Your curves look healthy - both training and validation losses are decreasing together without signs of overfitting (where validation loss would start increasing while training loss continues dropping). The [loss function documentation](https://www.ultralytics.com/glossary/loss-function) explains the distinction between training losses and evaluation metrics in more detail.

Yes, the metrics are averaged across all classes by default, giving you an overall performance view of your model.
